GraphWebApi/Controllers/GraphExplorerAppModeController.cs

```diff
@@ -2,60 +2,185 @@
+
+        public GraphExplorerAppModeController(IConfiguration configuration, ITokenAcquisition tokenAcquisition, IGraphAppAuthProvider graphServiceClient)
+        {
+            this._graphClient = graphServiceClient;
+            this._tokenAcquisition = tokenAcquisition;
+            this._config = configuration;
+        }
         [Route("api/[controller]/{*all}")]
         [Route("graphproxy/{*all}")]
         [HttpGet]
-        public async Task<IActionResult> GetAsync(string all, [FromHeader] string Authorization)
+        [AuthorizeForScopes(Scopes = new[] { "https://graph.microsoft.com/.default" })]
+        public async Task<IActionResult> GetAsync(string all, string tenantId)
         {
-            return await ProcessRequestAsync("GET", all, null, Authorization).ConfigureAwait(false);
+            return await this.ProcessRequestAsync("GET", all, null, _config.GetSection("AzureAd").GetSection("TenantId").Value).ConfigureAwait(false);
         }
 
+        private async Task<string> GetTokenAsync(string tenantId)
+        {
+            // Acquire the access token.
+            string scopes = "https://graph.microsoft.com/.default";
+
+            return await _tokenAcquisition.GetAccessTokenForAppAsync(scopes, tenantId, null);
+        }
+
+
         [Route("api/[controller]/{*all}")]
         [Route("graphproxy/{*all}")]
         [HttpPost]
-        public async Task<IActionResult> PostAsync(string all, [FromBody] object body, [FromHeader] string Authorization)
+        [AuthorizeForScopes(Scopes = new[] { "https://graph.microsoft.com/.default" })]
+        public async Task<IActionResult> PostAsync(string all, [FromBody] object body, string tenantId)
         {
-            return await ProcessRequestAsync("POST", all, body, Authorization).ConfigureAwait(false);
+            return await ProcessRequestAsync("POST", all, body, _config.GetSection("AzureAd").GetSection("TenantId").Value).ConfigureAwait(false);
         }
 
         [Route("api/[controller]/{*all}")]
         [Route("graphproxy/{*all}")]
         [HttpDelete]
-        public async Task<IActionResult> DeleteAsync(string all, [FromHeader] string Authorization)
+        public async Task<IActionResult> DeleteAsync(string all, string tenantId)
         {
-            return await ProcessRequestAsync("DELETE", all, null, Authorization).ConfigureAwait(false);
+            return await ProcessRequestAsync("DELETE", all, null, _config.GetSection("AzureAd").GetSection("TenantId").Value).ConfigureAwait(false);
         }
 
         [Route("api/[controller]/{*all}")]
         [Route("graphproxy/{*all}")]
         [HttpPut]
-        public async Task<IActionResult> PutAsync(string all, [FromBody] object body, [FromHeader] string Authorization)
+        [AuthorizeForScopes(Scopes = new[] { "https://graph.microsoft.com/.default" })]
+        public async Task<IActionResult> PutAsync(string all, [FromBody] object body, string tenantId)
```

Comment:
Gotta remove tenantId from the parameter of every function

Reply:
b3869bd0cdfa9022066be130e46c90538e78c65b